python/lsst/sims/coordUtils/CameraUtils.py

```diff
@@ -1,33 +1,204 @@
+
+    if are_arrays:
+        n_pts = len(input_list[0])
+    else:
+        n_pts = 1
+
+    if isinstance(chip_name, list) or isinstance(chip_name, np.ndarray):
+        if len(chip_name) > 1 and len(chip_name) != n_pts:
+            raise RuntimeError("You passed %d chipNames to %s.\n" % (len(chip_name), method_name) +
+                               "You passed %d %s values." % (len(input_list[0]), input_names[0]))
+
+        if len(chip_name) == 1 and n_pts > 1:
+            chip_name_out = [chip_name[0]]*n_pts
+        else:
+            chip_name_out = chip_name
+
+        return are_arrays, chip_name_out
+
+    elif chip_name is None:
+        return are_arrays, chip_name
+    else:
+        return are_arrays, [chip_name]*n_pts
+
+
+def getCornerPixels(detector_name, camera):
+    """
+    Return the pixel coordinates of the corners of a detector.
+
+    @param [in] detector_name is the name of the detector in question
+
+    @param [in] camera is the afwCameraGeom camera object containing
+    that detector
+
+    @param [out] a list of tuples representing the (x,y) pixel coordinates
+    of the corners of the detector.  Order will be
+
+    [(xmin, ymin), (xmin, ymax), (xmax, ymin), (xmax, ymax)]
+    """
+
+    det = camera[detector_name]
+    bbox = det.getBBox()
+    xmin = bbox.getMinX()
+    xmax = bbox.getMaxX()
+    ymin = bbox.getMinY()
+    ymax = bbox.getMaxY()
+    return [(xmin, ymin), (xmin, ymax), (xmax, ymin), (xmax, ymax)]
+
+
+def getCornerRaDec(detector_name, camera, obs_metadata, epoch=2000.0,
+                   includeDistortion=True):
+    """
+    Return the ICRS RA, Dec values of the corners of the specified
+    detector in degrees.
+
+    @param [in] detector_name is the name of the detector in question
+
+    @param [in] camera is the afwCameraGeom camera object containing
+    that detector
+
+    @param [in] obs_metadata is an ObservationMetaData characterizing
+    the pointing (and orientation) of the telescope.
+
+    @param [in] epoch is the mean Julian epoch of the coordinate system
+    (default is 2000)
+
+    @param [in] includeDistortion is a boolean.  If True (default), then this method will
+    convert from pixel coordinates to RA, Dec with optical distortion included.  If False, this
+    method will use TAN_PIXEL coordinates, which are the pixel coordinates with
+    estimated optical distortion removed.  See the documentation in afw.cameraGeom for more
+    details.
+
+    @param [out] a list of tuples representing the (RA, Dec) coordinates
+    of the corners of the detector in degrees.  The corners will be
+    returned in the order
+
+    [(xmin, ymin), (xmin, ymax), (xmax, ymin), (xmax, ymax)]
+
+    where (x, y) are pixel coordinates.  This will not necessarily
+    correspond to any order in RAmin, RAmax, DecMin, DecMax, because
+    of the ambiguity imposed by the rotator angle.
+    """
+
+    cc = _getCornerRaDec(detector_name, camera, obs_metadata,
```

Comment:
Just wondering. Is the reason you have this as two methods so that if you wanted to do something with the ra/dec coordinates you wouldn't have to call this method and then switch back to radians for some sort of transformation?